risc0/bigint2/src/ec/mod.rs

```diff
@@ -158,36 +195,36 @@ impl<const WIDTH: usize, C: Curve<WIDTH>> AffinePoint<WIDTH, C> {
         let curve = C::CURVE;
 
         // If the left or right value is zero, can return the other value.
-        if self.infinity {
+        let Some(lhs) = self.as_u32s() else {
             *result = *rhs;
-        } else if rhs.infinity {
+            return;
+        };
+        let Some(rhs) = rhs.as_u32s() else {
             *result = *self;
-        } else if self.buffer[0] == rhs.buffer[0] {
+            return;
+        };
+
+        if lhs[0] == rhs[0] {
             // X coordinates are the same, so either we double the value if it's the same point,
             // or return the identity if it's different (not on the curve).
```

Comment:
An assumption of this function is that both points are on the curve; so if the points are different but have the same x value they must be negatives (because if (x, y) is on the curve, the only other point on the curve with the same x coordinate is (x, -y) which is -(x, y) under the elliptic curve group law).
```suggestion
            // or return the identity if it's different (and so lhs = -rhs).
```